app/bucket_list.py

```diff
@@ -0,0 +1,198 @@
+            else:
+                prev_page = 'None'
+
+            if bucketlists.has_next:
+                next_page = request.url + '?page=' + str(page + 1) + '&limit=' + str(limit)
+            else:
+                next_page = 'None'
+
+            return {
+                'message': {
+                    'next_page':next_page,
+                    'prev_page': prev_page,
+                    'total': bucketlists.pages},
+                'bucketlists': marshal(bucketlists.items, bucket_list_fields)
+            }, 200
+
+
+    def post(self):
+        """ Create a new bucket list """
+        args = self.reqparse.parse_args()
+        name = args['bucketlist_name']
+
+        check_duplicate = BucketList.query.filter_by(bucketlist_name=name).first()
+
+        if check_duplicate:
+            return {'error': 'bucketlist_name {} already exists'.format(name)}
+
+        new_bucketlist = BucketList(bucketlist_name=name, created_by=g.user.user_id)
+        db.session.add(new_bucketlist)
+        db.session.commit()
+
+        return marshal(new_bucketlist, bucket_list_fields)
+
+    def put(self, id):
+        """ Update a bucket list """
+        this_bucket_list = BucketList.query.filter_by(bucketlist_id=id).first()
+        if not this_bucket_list:
+            return {'error': 'bucketlist with id {} does not exists'.format(id)}
+        args = self.reqparse.parse_args()
+        name = args['bucketlist_name']
+
+        update_bucketlist = BucketList.query.filter_by(bucketlist_id=id).first()
+        update_bucketlist.bucketlist_name = name
+        db.session.commit()
+
+        return marshal(update_bucketlist, bucket_list_fields)
+
+    def delete(self, id):
+        """ Delete a single bucket list """
+        this_bucket_list = BucketList.query.filter_by(bucketlist_id=id).first()
+        if not this_bucket_list:
+            return {'error': 'bucketlist with id {} does not exists'.format(id)}
+
+        db.session.delete(this_bucket_list)
+        db.session.commit()
+
+        return {'message': 'bucketlist with id {} has been deleted'.format(id)}
+
+class ItemAPI(Resource):
+    """ Creates endpoints for ItemAPI """
+    decorators = [auth_user.login_required]
+
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('item_name', type=str, required=True, \
+        help='No Item name provided', location='json')
+        super(ItemAPI, self).__init__()
+
+    def post(self, bucketlist_id):
+        """ Create a new item in bucket list based on the bucketlist_id """
+        args = self.reqparse.parse_args()
+        name = args['item_name']
+        bucketlist_exists = BucketList.query.filter_by(bucketlist_id=bucketlist_id).first()
+
+        if not bucketlist_exists:
+            return {'error': 'Bucketlist_id {} does not exists'.format(bucketlist_id)}
+
+        new_item = Item(item_name=name, bucketlist_id=bucketlist_id)
+        db.session.add(new_item)
```

Comment:
In L155 consider the permission issue of filtering bucketlist of current user. L160 What if a user creates an bucketlist item with same name?